wagtail_localize/locales/tests.py

```diff
@@ -1,61 +1,124 @@
 
-    def get(self, params=None):
-        return self.client.get(reverse("wagtaillocales:add"), params or {})
+        self.assertIn("wagtail_version", response.context)
+        self.assertIsInstance(response.context["wagtail_version"], str)
+
+    def test_queryset_filtering(self):
+        response = self.execute_request("GET", "wagtaillocales:index")
+        self.assertEqual(response.status_code, 200)
 
+        # Check that all objects in the context belong to the expected queryset
+        for locale in response.context["locales"]:
+            self.assertIn(locale, Locale.all_objects.all())
+
+    def test_page_title(self):
+        response = self.execute_request("GET", "wagtaillocales:index")
+        self.assertEqual(response.status_code, 200)
+
+        # Assume that page_title returns the expected value
+        expected_page_title = gettext_lazy("Locales")
+        self.assertEqual(response.context["page_title"], expected_page_title)
+
+    def test_wagtail_version(self):
+        response = self.execute_request("GET", "wagtaillocales:index")
+        self.assertEqual(response.status_code, 200)
+
+        # Assume that get_main_version returns the expected value
+        expected_wagtail_version = get_main_version()
+        self.assertEqual(response.context["wagtail_version"], expected_wagtail_version)
+
+    def test_get_locale_usage(self):
+        # Test get_locale_usage function with different scenarios
+        # For example, test with a locale that has no pages and others
+
+        # Create a unique locale for testing
+        locale = Locale.all_objects.create(language_code="fr")
+        self.assertEqual(locale.language_code, "fr")
+
+        # Execute the request and check if the locale is present in the context
+        response = self.execute_request("GET", "wagtaillocales:index")
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(locale, response.context["locales"])
+
+
+class TestLocaleCreateView(BaseLocaleTestCase):
     def post(self, post_data=None):
+        # Helper method for making POST requests to create a locale
         return self.client.post(reverse("wagtaillocales:add"), post_data or {})
 
     def test_default_language(self):
-        # we should have loaded with a single locale
+        # Ensure the default language is set up correctly
         self.assertEqual(self.english.language_code, "en")
         self.assertEqual(self.english.get_display_name(), "English")
 
     def test_simple(self):
-        response = self.get()
+        # Test if the create view renders successfully with correct choices
+        response = self.client.get(reverse("wagtaillocales:add"))
         self.assertEqual(response.status_code, 200)
         self.assertTemplateUsed(response, "wagtaillocales/create.html")
-
         self.assertEqual(
             response.context["form"].fields["language_code"].choices, [("fr", "French")]
         )
 
     def test_create(self):
-        response = self.post(
```

Comment:
`TestLocaleCreateView` still has a `post` method that posts to `reverse("wagtaillocales:add")`. I'd say keep using it, especially that we're using that in at least one other place